In [1]:
#import necessity libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
import seaborn as sns
import statsmodels.api as smf
import warnings
warnings.filterwarnings('ignore')

Question no.1

The time required for servicing transmissions is normally distributed with  = 45 minutes and  = 8 minutes. The service manager plans to have work begin on the transmission of a customer’s car 10 minutes after the car is dropped off and the customer is told that the car will be ready within 1 hour from drop-off. What is the probability that the service manager cannot meet his commitment?

#Let 'X' be the amount of time it takes to complete the repair on a customer's car. So the Probability of completing the car repair with 1 hr of dropping the car and starting the work after 10 mins of car drop (60 mins - 10 mins) is P(X<=50 mins) So, P(X>50) = 1-P(X<=50) i.e. cannot meet his commitment


In [16]:
# Find Z-Scores at X=50; Z = (X - µ) / σ 
Z=(50-45)/8
Z

0.625

In [17]:
# Find probability P(X>50) = 1-stats.norm.cdf(abs(z_score))
1-stats.norm.cdf(abs(0.625))

0.26598552904870054

In [18]:
# OR Find probability P(X<=50); p_value=stats.norm.cdf(abs(z_score))  
p_value=stats.norm.cdf(abs(0.625)) 
p_value

0.7340144709512995

In [19]:
# P(X>50) = 1 - P(X<=50)
1-0.734

0.266

Therefore, the answer is b.

Question no.2

The current age (in years) of 400 clerical employees at an insurance claims processing center is normally distributed with mean  = 38 and Standard deviation  =6. For each statement below, please specify True/False.

In [4]:
mean = 38
std1 = 6
q2_lessthan_38 = stats.norm.cdf(38, loc = mean, scale = std1)


q2_less_than_44 = stats.norm.cdf(44, loc = mean,  scale = std1)


q2_betweeen_38_and_44 = (q2_less_than_44 - q2_lessthan_38)
print('The probability of employee age betweeen 38 and 44 is',np.round(q2_betweeen_38_and_44*100,2),'%') 

q2_morethan_44 = 1-stats.norm.cdf(44, loc = mean, scale = std1)
print('The probability of employee age more than 44 is',np.round(q2_morethan_44*100,2),'%')

true_or_false = (q2_morethan_44 > q2_betweeen_38_and_44)
print('Answer:',true_or_false)

q2b = stats.norm.cdf(30, loc = mean, scale = std1)
print("""A training program for employees under the age of 30 at the center would be expected to attract about"""
      ,np.round((q2b*400),0),'employees')

The probability of employee age betweeen 38 and 44 is 34.13 %
The probability of employee age more than 44 is 15.87 %
Answer: False
A training program for employees under the age of 30 at the center would be expected to attract about 36.0 employees


Question no.4

Let X ~ N (100, 202). Find two values, a and b, symmetric about the mean, such that the probability of the random variable taking a value between them is 0.99.

In [25]:
stats.norm.interval(0.99,100,20)

(48.48341392902199, 151.516586070978)

a = 48.48 and b = 151.52

Question no.5

Consider a company that has two different divisions. The annual profits from the two divisions are independent and have distributions Profit1 ~ N(5, 32) and Profit2 ~ N(7, 42) respectively. Both the profits are in USD Million. Answer the following questions about the total profit of the company in Rupees. Assume that 1USD = Rs. 45

In [20]:
# Mean profits from two different divisions of a company = Mean1 + Mean2
Mean = 5+7
print('Mean Profit is Rs', Mean*45,'Million')

Mean Profit is Rs 540 Million


In [21]:
# Variance of profits from two different divisions of a company = SD^2 = SD1^2 + SD2^2
SD = np.sqrt((9)+(16))
print('Standard Deviation is Rs', SD*45, 'Million')

Standard Deviation is Rs 225.0 Million


A. Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.

In [22]:
print('Range is Rs',(stats.norm.interval(0.95,540,225)),'in Millions')

Range is Rs (99.00810347848784, 980.9918965215122) in Millions


B. Specify the 5th percentile of profit (in Rupees) for the company

In [23]:
# To compute 5th Percentile, we use the formula X=μ + Zσ; wherein from z table, 5 percentile = -1.645
X= 540+(-1.645)*(225)
print('5th percentile of profit (in Million Rupees) is',np.round(X,))

5th percentile of profit (in Million Rupees) is 170.0


C. Which of the two divisions has a larger probability of making a loss in a given year?

In [24]:
# Probability of Division 2 making a loss P(X<0)
stats.norm.cdf(0,7,4)

0.040059156863817086

Inference: Probability of Division 1 making a loss in a given year is more than Division 2.